### ISSUE TITLE
Dell 5450/5550 Issues

### ISSUE THREAD

**Issue by FlannelNZ** [maurice-daly/DriverAutomationTool#538]
Crossposting, as its probably more relevant here.

The Automation Tool is bringing down the Old E5450/E5550 driver packs (A08/A09) rather than the new Latitude 5450/5550 A00 CAB's from Dell.

**Comment by FlannelNZ** [comment]
![image](https://github.com/maurice-daly/DriverAutomationTool/assets/67209612/bdd5ae75-b1d7-4760-a555-19f3e797f693)
Despite clicking the Latitude 5550, It actually brings down E5550/5550 which is not the correct cab.

**Comment by Azzaro78** [comment]
Having near to the same issue. 5450 isn't even downloading the cab:
![2024-04-16_11-40](https://github.com/maurice-daly/DriverAutomationTool/assets/46527526/e4c106d9-d5a5-407b-a1af-b8bba5bd93a3)

Other driver packs work. So it does indeed looks like the script is confused between E5450 and 5450 and Windows 11.

**Comment by WilhelmMetzger** [comment]
We have the same problems on 5350/5450/5550. Does anyone have a solution to this problem or a workaround? Have any of you reported this to Dell?

**Comment by FlannelNZ** [comment]
The workaround is to manually download the correct package and create a custom package for each model as per the Driver Automation Tool Documentation and then unselect those models in the make and model selection page.

Dell would probably be uninterested. And any change to the naming of their driver package would likely take an age. But it might be worth raising is someone has the ear of someone in their Enterprise team.

**Comment by browndervilleb** [comment]
Same Issue Here.. Took quite a while to figure this was the issue.

**Comment by Acabar** [comment]
Does anyone have the baseboards handy for these models? It won't let me create the custom package without them and we don't yet have the physical devices available.

**Comment by FlannelNZ** [comment]
I presume you are referring to the SKU.
We have some already, here they are:

5450 (Models included;0CB2;0CB4)
5550 (Models included:0CB9;0CBC)

**Comment by Acabar** [comment]
> I presume you are referring to the SKU. We have some already, here they are:
> 
> 5450 (Models included;0CB2;0CB4) 5550 (Models included:0CB9;0CBC)

Thanks! There's also a baseboard number the tool asks for that can be pulled using the following powershell command on a device: `(Get-CIMInstance -ClassName MS_SystemInformation -NameSpace root\WMI).BaseBoardProduct`

**Comment by smaragos64** [comment]
This is an issue with the Dell XML. The driver package data for the Latitude E5450 is missing the E within SupportedSystems. If you review the DriverautomationTool.log you can see the incorrect additional Dell System Model ID
- Dell System Model ID is : 0645;0CB2;0CB4
The Model ID of 0645 is for the Latitude E5450. 
With this incorrect data, multiple other parts of the tool break since it uses the first value in the list. 

I was able to manually modify the DriverPackCatalog.xml to correct this mistake and the tool was able to download and build the Latitude 5450.  I believe there are other models in a similarly broken state.

**Comment by Acabar** [comment]
> This is an issue with the Dell XML. The driver package data for the Latitude E5450 is missing the E within SupportedSystems. If you review the DriverautomationTool.log you can see the incorrect additional Dell System Model ID
> 
> * Dell System Model ID is : 0645;0CB2;0CB4
>   The Model ID of 0645 is for the Latitude E5450.
>   With this incorrect data, multiple other parts of the tool break since it uses the first value in the list.
> 
> I was able to manually modify the DriverPackCatalog.xml to correct this mistake and the tool was able to download and build the Latitude 5450. I believe there are other models in a similarly broken state.

Awesome find! I had done some looking into the XML but wasn't sure 100% where to begin, but this makes perfect sense. It looks like you can also tell which is the old E- series by comparing the rstDate= parameter as well.

**Comment by browndervilleb** [comment]
> This is an issue with the Dell XML. The driver package data for the Latitude E5450 is missing the E within SupportedSystems. If you review the DriverautomationTool.log you can see the incorrect additional Dell System Model ID
> 
> * Dell System Model ID is : 0645;0CB2;0CB4
>   The Model ID of 0645 is for the Latitude E5450.
>   With this incorrect data, multiple other parts of the tool break since it uses the first value in the list.
> 
> I was able to manually modify the DriverPackCatalog.xml to correct this mistake and the tool was able to download and build the Latitude 5450. I believe there are other models in a similarly broken state.

So you just need the "E" for each of the names that show up when I search for systemID="0645" ?
Before: <Model systemID="0645" name="Latitude 5450" rtsDate="2014-09-14T00:00:00" generation="X6"> 
After: <Model systemID="0645" name="Latitude E5450" rtsDate="2014-09-14T00:00:00" generation="X6">

**Comment by smaragos64** [comment]
Yes, adding that should resolve the Latitude 5450. I believe it is there 2 times, once each for Windows 10 and Windows 11.

**Comment by FlannelNZ** [comment]
Good spotting. I started looking through there, but it was a bit overwhelming. 
I have raised this via our Account Exec, hopefully he can put me on to the Enterprise CAB team.

**Comment by smaragos64** [comment]
I was able to successfully build the Latitude 5550 with these steps

1) Delete any cab and xml files in the Driver Automation Tool\Temp  directory
2) Launch the tool to download the latest versions
3) Modify the xml as mentioned
4) Close and relaunch the tool
5) Verify the xml file is still the modified version
6) Start the download

**Comment by Aron-Love** [comment]
FWIW, I brought this up during my school district's weekly Dell meeting. The tech on the call said the real fix is to use the System Model ID, which will always be different. But he will bring it up with a contact on the team that deals with the XML. Who knows when it will get updated, though?

**Comment by kirderf85** [comment]
> I was able to successfully build the Latitude 5550 with these steps
> 
> 1. Delete any cab and xml files in the Driver Automation Tool\Temp  directory
> 2. Launch the tool to download the latest versions
> 3. Modify the xml as mentioned
> 4. Close and relaunch the tool
> 5. Verify the xml file is still the modified version
> 6. Start the download

I just tested and can confirm that the steps above worked. Thanks!

**Comment by Chathushka34** [comment]
Due to the DAT issue we pulled drivers manually and added to task sequence. But after imaging to Win 11 and run Dell command update, it completely disable NIC drivers. Does anybody facing this issue? I'm thinking Dell command update pulls old drivers for latitude 5550/5450 machines.
